packages/features/insights/server/routing-events.ts

```diff
@@ -0,0 +1,459 @@
+  fieldFilter?: {
+    fieldId: string;
+    optionId: string;
+  } | null;
+};
+
+class RoutingEventsInsights {
+  private static async getWhereForTeamOrAllTeams({
+    teamId,
+    isAll,
+    organizationId,
+    routingFormId,
+  }: RoutingFormInsightsTeamFilter) {
+    // Get team IDs based on organization if applicable
+    let teamIds: number[] = [];
+    if (isAll && organizationId) {
+      const teamsFromOrg = await prisma.team.findMany({
+        where: {
+          parentId: organizationId,
+        },
+        select: {
+          id: true,
+        },
+      });
+      teamIds = [organizationId, ...teamsFromOrg.map((t) => t.id)];
+    } else if (teamId) {
+      teamIds = [teamId];
+    }
+
+    // Base where condition for forms
+    const formsWhereCondition: Prisma.App_RoutingForms_FormWhereInput = {
+      ...(teamIds.length > 0 && {
+        teamId: {
+          in: teamIds,
+        },
+      }),
+      ...(routingFormId && {
+        id: routingFormId,
+      }),
+    };
+
+    return formsWhereCondition;
+  }
+
+  static async getRoutingFormStats({
+    teamId,
+    startDate,
+    endDate,
+    isAll = false,
+    organizationId,
+    routingFormId,
+    userId,
+    bookingStatus,
+    fieldFilter,
+  }: RoutingFormInsightsFilter) {
+    // Get team IDs based on organization if applicable
+    const formsWhereCondition = await this.getWhereForTeamOrAllTeams({
+      teamId,
+      isAll,
+      organizationId,
+      routingFormId,
+    });
+
+    // Base where condition for responses
+    const responsesWhereCondition: Prisma.App_RoutingForms_FormResponseWhereInput = {
+      ...(startDate &&
+        endDate && {
+          createdAt: {
+            gte: dayjs(startDate).startOf("day").toDate(),
+            lte: dayjs(endDate).endOf("day").toDate(),
+          },
+        }),
+      ...(userId || bookingStatus
+        ? {
+            ...(bookingStatus === "NO_BOOKING"
+              ? { routedToBooking: null }
+              : {
+                  routedToBooking: {
+                    ...(userId && { userId }),
+                    ...(bookingStatus && { status: bookingStatus }),
+                  },
+                }),
+          }
+        : {}),
+      ...(fieldFilter && {
+        response: {
+          path: [fieldFilter.fieldId, "value"],
+          array_contains: [fieldFilter.optionId],
+        },
+      }),
+      form: formsWhereCondition,
+    };
+
+    // Get total forms count
+    const totalForms = await prisma.app_RoutingForms_Form.count({
+      where: formsWhereCondition,
+    });
+
+    // Get total responses
+    const totalResponses = await prisma.app_RoutingForms_FormResponse.count({
+      where: responsesWhereCondition,
+    });
+
+    // Get responses without booking
+    const responsesWithoutBooking = await prisma.app_RoutingForms_FormResponse.count({
+      where: {
+        ...responsesWhereCondition,
+        routedToBookingUid: null,
+      },
+    });
```

Comment:
I see three async calls that don't rely on each other. Can we wrap this in a `Promise.all()`?